### PR TITLE
Interfacification fixes 2

### PR DESCRIPTION
**Pull request by tal5**
## Additions

- `BukkitScriptEvent#couldMatchRegistry` - couldMatch checker for checking against a registry.
- `EntityHelper#getBlockHeight/Width` - for calling the equivalent methods on `Art` across versions.
- `server.sound_keys` - returns modern Mojang sound names, on 1.20+.
- `Utilities#listLegacyTypes` - like `listTypes`, but to get the older legacy names in cases where the keys and Spigot names don't match (e.g. sounds).
- `Utilities#enumLikeToLegacyElement` - same as `enumlikeToElement`, but ^ legacy names always.

## Changes

- `NoteBlockPlaysNoteScriptEvent#getSound` now uses the new `Instrument#getSound` on relevant versions.
- Changed relevant places to use the enum compatibility methods.
- `BiomeEnterExitScriptEvent` now uses `BukkitScriptEvent#couldMatchRegistry`.
- The `BiomeTag` `Biome` constructor now casts to `Keyed` when calling `biome.getKey` to make older versions accept the `INVOKEINTERFACE`.
- `EntityTag` painting special handling now uses the new `EntityHelper` methods and `Registry.ART` (present on all supported versions).
- `EntityPainting` & `PlaySoundCommand` updated to relevant cross-version support methods.
- `server.biome_types` now uses `Registry.BIOME` (present on all supported versions) & updated meta to reflect it now returning custom biomes.
- `server.sound_types` now uses the appropriate cross-version method, and is deprecated in favor of `server.sound_keys` on MC 1.21+ with the general `oldSpigotNames` deprecation warning.
- The `SoundLookup` util was deleted in favor of just using the sound registry (present on all supported versions).

### CODE REVIEW

File: plugin/src/main/java/com/denizenscript/denizen/events/BukkitScriptEvent.java [hunk 1]
```diff
@@ -163,6 +163,22 @@ public static boolean couldMatchEnum(String text, final Enum<?>[] enumVals) {
         return false;
     }
 
+    public static boolean couldMatchRegistry(String text, Registry<?> registry) {
+        String textLower = CoreUtilities.toLowerCase(text);
+        MatchHelper matcher = isAdvancedMatchable(text) ? createMatcher(text) : null;
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/events/BukkitScriptEvent.java]: this alternative handling is redundant the way this is implemented. Could swap the non-matchable version to be a `registry.get` call, or just drop the ternary and always use matcher

File: plugin/src/main/java/com/denizenscript/denizen/utilities/Utilities.java [hunk 2]
```diff
@@ -562,6 +568,17 @@ public static ElementTag enumlikeToElement(Object val) {
...
+        if (val instanceof Enum<?> enumVal) {
+            return new ElementTag(enumVal);
+        }
+        if (val instanceof OldEnum<?> oldEnumVal) {
+            return new ElementTag(oldEnumVal.name(), true);
+        }
+        throw new UnsupportedOperationException("Cannot get legacy name element, value isn't an enum: " + val);
+    }
+
+    // TODO: need proper input backsupport, see discord://discord.com/channels/315163488085475337/1011496047811506227/1301272242386370580
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/utilities/Utilities.java]: `https` not magic unsupported discord protocol

**Comment by tal5** [comment]
Should be ready for rereview

**Review by mcmonkey4eva — APPROVED** [review]

**Comment by mcmonkey4eva** [comment]
all looks good to me

**Pull request merged**